proto/model.proto

```diff
@@ -0,0 +1,151 @@
+syntax = "proto2";
```

Comment:
if we are updating the existing model to be faster & efficient why not also move to proto3?

Reply:
The main difference between proto3 and proto2 is that in proto3 a property is always initialized with their default value (a.k.a. no pointer used for every member). By using gogoproto + `(gogoproto.nullable) = false` you achieve the same result.

The main reason not to change is the use of `required` in the `Event`, in proto3 this was removed so if ServiceA(no required) calls ServiceB (uses required) and the `eventType` is empty the request will fail. If we want to safely change to proto3 (which again will not bring any benefit for gogoproto, but for other languages will be a win) we need to first remove the use of `required`, which can be done by changing all the backends (from bottom to the top) to not use `required` for that field.

Reply:
Also I had the initial version using proto3 and I cannot see any significant difference in benchmarks.

Reply:
And to keep in mind the generated code is the same for gogoproto, so we can always change to proto3 (without having to change the code that consumes the generated code) once the `required` problem is fixed.

Reply:
ack